src/App.js

```diff
@@ -21,45 +21,113 @@ const generateSquares = () => {
       currentId += 1;
     }
   }
-
   return squares;
 };
 
 const App = () => {
-  // This starts state off as a 2D array of JS objects with
-  // empty value and unique ids.
   const [squares, setSquares] = useState(generateSquares());
+  const [player, setPlayer] = useState(PLAYER_1);
+  const [winner, setWinner] = useState('...');
 
-  // Wave 2
-  // You will need to create a method to change the square
-  //   When it is clicked on.
-  //   Then pass it into the squares as a callback
+  const onSquareClickCallback = (id) => {
+    const boardCopy = squares.map((row) => {
+      row.forEach((column) => {
+        if (column.id === id) {
+          if (column.value === '') {
+            if (player === PLAYER_1) {
+              column.value = PLAYER_1;
+            } else {
+              column.value = PLAYER_2;
+            }
+          }
+        }
+      });
+      return row;
+    });
+    setSquares(boardCopy);
+    if (player === PLAYER_1) {
+      setPlayer(PLAYER_2);
+    } else {
+      setPlayer(PLAYER_1);
+    }
+
+    checkForWinner();
+  };
 
   const checkForWinner = () => {
```

Comment:
Consider additional logic that prevents the game from continuing once someone has won.